### PR TITLE
samples: lvgl: reduce footprint

### PR DESCRIPTION
**Pull request by kartben**
prj.conf was manually "enabling" widgets giving a false impression of fine tuning, but they were in fact already enabled anyway, alongside a lot of other unused widgets and features.

Switched to CONFIG_LV_CONF_MINIMAL and added a Kconfig to go even further in enabling only widgets that are strictly needed.

This is still not aiming for a "minimal" footprint (ex. we still pull monkey, logging, a fancy font...) but this reduces flash usage by roughly 80KB nevertheless :)

M5 Stack Core2 (has touchscreen):
- before: 285KB
- after: 206KB

Wio Terminal (no touchscreen):
- before: 255KB
- after: 179KB

### CODE REVIEW

**Review by faxe1008 — COMMENTED** [review]
Makes sense that the basic sample is as optimized as possible to also be usable on smaller devices. What do you think about applying this also to your other PR, or do you think that the accelerometer sample needs a more performant device anyways?

**Comment by kartben** [comment]
> Makes sense that the basic sample is as optimized as possible to also be usable on smaller devices.

Ya but tricky too. I.e. fully optimized would also mean ugly, as we'd want to not use the default theme, not use Montserrat and just the not so pretty system font, etc. 

> What do you think about applying this also to your other PR, or do you think that the accelerometer sample needs a more performant device anyways?

Mmmh yeah let me try and update it

**Comment by fabiobaltieri** [comment]
> > Makes sense that the basic sample is as optimized as possible to also be usable on smaller devices.
> 
> Ya but tricky too. I.e. fully optimized would also mean ugly, as we'd want to not use the default theme, not use Montserrat and just the not so pretty system font, etc.

Would it make sense to have some sort of "minimal" setup of the test with an overlay or something?

**Comment by kartben** [comment]
> > > Makes sense that the basic sample is as optimized as possible to also be usable on smaller devices.
> > 
> > 
> > Ya but tricky too. I.e. fully optimized would also mean ugly, as we'd want to not use the default theme, not use Montserrat and just the not so pretty system font, etc.
> 
> Would it make sense to have some sort of "minimal" setup of the test with an overlay or something?

Ya I like the idea. I think it's important that people using the basic LVGL sample as a starting point can both see it work in a minimal setup, but also have an easy way to extend the sample (ex. add new types of widgets) without getting stuck due to features being disabled. I.e. for the former an overlay setting the minimal config options would be used, but for the latter folks would be encouraged to use the more regular setup.

**Review by jfischer-no — APPROVED** [review]

File: samples/subsys/display/lvgl/Kconfig [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#
+# Copyright (c) 2023 Benjamin Cabé <benjamin@zephyrproject.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
```
Comment (faxe1008) [inline samples/subsys/display/lvgl/Kconfig:6]: @kartben This PR has been sitting around for a bit now  - sorry for that. Can you maybe add the conditional select for the `button_matrix` based on `LV_Z_KEYPAD_INPUT`?

Comment (kartben) [reply samples/subsys/display/lvgl/Kconfig:6]: No worries at all! I will try to update but TBH it also feels like reinstating a "minimal" hello world would also be beneficial now that the original sample has grown into a somewhat rich input devices showcase

**Pull request closed without merging**